plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -1207,6 +1207,17 @@ protected void execute(ConnectorSession session, Connection connection, String q
         }
     }
 
+    protected void execute(ConnectorSession session, Connection connection, List<String> queries)
+            throws SQLException
+    {
+        connection.setAutoCommit(false);
+        for (String query : queries) {
+            verify(!connection.getAutoCommit());
+            execute(session, connection, query);
+        }
+        connection.commit();
```

Comment:
I am not sure if we should control transactions that explicit here. I think this should handled outside of this method. Also I mentioned in other comment to possibly inline this method.